util/store/store.go

```diff
@@ -106,155 +96,101 @@ func SetObject(store sdk.KVStore, cdc codec.Codec, key []byte, object codec.Prot
-	return setInteger[int64](store, key, &gogotypes.Int64Value{Value: val}, errField)
-}
-
 // GetAddress retrieves an sdk.AccAddress from a KVStore, or an empty address if no value is stored.
-// Panics if a non-empty address fails sdk.VerifyAddressFormat, so non-empty returns are always valid.
 // Accepts an additional string which should describe the field being retrieved in custom error messages.
-func GetAddress(store sdk.KVStore, key []byte, errField string) sdk.AccAddress {
+func GetAddress(store sdk.KVStore, key []byte) sdk.AccAddress {
 	if bz := store.Get(key); len(bz) > 0 {
 		addr := sdk.AccAddress(bz)
-		if err := sdk.VerifyAddressFormat(addr); err != nil {
-			panic(fmt.Sprintf("%s is not valid: %s", errField, err))
-		}
 		return addr
 	}
 	// No stored bytes at key: return empty address
 	return sdk.AccAddress{}
 }
 
 // SetAddress stores an sdk.AccAddress in a KVStore, or clears if setting to an empty or nil address.
-// Returns an error on attempting to store a non-empty address that fails sdk.VerifyAddressFormat.
 // Accepts an additional string which should describe the field being set in custom error messages.
-func SetAddress(store sdk.KVStore, key []byte, val sdk.AccAddress, errField string) error {
+func SetAddress(store sdk.KVStore, key []byte, val sdk.AccAddress) {
 	if val == nil || val.Empty() {
 		store.Delete(key)
-		return nil
-	}
-	if err := sdk.VerifyAddressFormat(val); err != nil {
-		return fmt.Errorf("%s is not valid: %s", errField, err)
+		return
 	}
 	store.Set(key, val)
-	return nil
 }
 
-func getInteger[Num constraints.Integer, TPtr gogoInteger[T, Num], T any](
-	store sdk.KVStore, key []byte, errField string) Num {
-
-	v := GetValue[TPtr](store, key, errField)
-	if v == nil {
-		// No stored bytes at key: return zero
-		return 0
+func SetInteger[Num Integer](store sdk.KVStore, key []byte, v Num) {
+	var bz []byte
+	switch v := any(v).(type) {
+	case int64:
+		bz = make([]byte, 8)
+		binary.LittleEndian.PutUint64(bz, uint64(v))
+	case uint64:
+		bz = make([]byte, 8)
+		binary.LittleEndian.PutUint64(bz, v)
+	case int32:
+		bz = make([]byte, 8)
+		binary.LittleEndian.PutUint32(bz, uint32(v))
+	case uint32:
+		bz = make([]byte, 8)
+		binary.LittleEndian.PutUint32(bz, v)
+	case byte:
+		bz = []byte{v}
 	}
-
-	return v.GetValue()
+	store.Set(key, bz)
 }
 
-func setInteger[Num constraints.Integer, TPtr gogoInteger[T, Num], T any](
-	store sdk.KVStore, key []byte, v TPtr, errField string) error {
-
-	if v.GetValue() == 0 {
-		store.Delete(key)
-		return nil
+func GetInteger[Num Integer](store sdk.KVStore, key []byte) Num {
+	bz := store.Get(key)
+	if bz == nil {
+		return 0
 	}
-	return SetValue(store, key, v, errField)
+	var v Num
+	switch any(v).(type) {
+	case int64, uint64:
+		v2 := binary.LittleEndian.Uint64(bz)
+		return Num(v2)
+	case int32, uint32:
+		return Num(binary.LittleEndian.Uint32(bz))
+	case byte:
+		return Num(bz[0])
+	}
+	panic("not possible!")
```

Comment:
Error message could be more specific here

Reply:
updated the message